### DISCUSSION TITLE
ESP32-S3 7inch Capacitive Touch Display

### DISCUSSION THREAD

**Discussion by DByhlin** [jeroen79/esp-brew-engine#47] [Ideas]
Any plans to make a version with both the web server and a touch interface? This screen is very reasonably priced, and seems to have great availability:

https://www.waveshare.com/esp32-s3-touch-lcd-7.htm

**Comment by jeroen79** [comment]
I personal use an old tablet to control the interface, its easier as i can place it where i want.

But i am not against trying to integrate a touchscreen, i can understand it easier in some setups, but as i do not have it lying around its quite difficult to support it atm.

**Reply by DByhlin** [reply]
I understand! Would be a fantastic way of replacing the control unit on a lot of older brewing systems, for example the oldest Braumeister units (Speidel wants about 800$ for a more modern control unit...) without loosing the "integrated" feel of the controller.

Think of it like an open source Brewtools control unit! :)

**Reply by jeroen79** [reply]
Yeah it would be quite nice, but it also needs to be seen if this screen has enough io's left over and enough flash rom to support both.

It might also just be an option to use a esp relay board for the controller and an extra esp with touchscreen as interface, that way the controller is not affected by any performance delays of the interface.

**Reply by DByhlin** [reply]
I'm ordering a display and a relay board to test if it's possible to make a LVGL UI on the display that communicates with the relay board over SPI. Just 2 months of delivery time to wait...

**Reply by kulikowskidesigns** [reply]
There are some very nice displays from 4D systems with integrated ESP32 micros and they have an IDE for designing the graphics.  I used to use their "diablo" version.  The only downside is they are a bit arduino-centric in terms of loading images etc.

https://4dsystems.com.au/products/gen4-esp32-series/

$125 for a 7 inch capacitive touchscreen including all the ESP32 stuff seems fairly reasonable.

**Comment by Praesil** [comment]
There is also a relatively simple to implent option:  I2C displays.  Personally, I would appreciate having a small OLED that I can configure to just display key statistics - Time, Temperature(s), heater status, etc. 

For control, phones, tablets, laptops, whatever - they're all fine.  Once the schedule is set, there's limited actual interaction needed with the software, but, having some targeted, limited feedback is very helpful.
